build.gradle

```diff
@@ -0,0 +1,354 @@
+                      "picard.vcf.GenotypeConcordance", "picard.illumina.IlluminaBasecallsToFastq", "picard.illumina.IlluminaBasecallsToSam", "picard.illumina.CheckIlluminaDirectory",
+                      "picard.sam.CheckTerminatorBlock", "picard.util.IntervalListTools", "picard.util.LiftOverIntervalList", "picard.vcf.LiftoverVcf", "picard.vcf.MakeSitesOnlyVcf",
+                      "picard.sam.markduplicates.MarkDuplicates", "picard.sam.markduplicates.MarkDuplicatesWithMateCigar", "picard.analysis.MeanQualityByCycle",
+                      "picard.sam.MergeBamAlignment", "picard.sam.MergeSamFiles", "picard.vcf.MergeVcfs", "picard.reference.NormalizeFasta", "picard.sam.PositionBasedDownsampleSam",
+                      "picard.reference.ExtractSequences", "picard.analysis.QualityScoreDistribution", "picard.vcf.RenameSampleInVcf", "picard.sam.ReorderSam",
+                      "picard.sam.ReplaceSamHeader", "picard.sam.RevertSam", "picard.sam.RevertOriginalBaseQualitiesAndAddMateCigar", "picard.sam.SamFormatConverter",
+                      "picard.sam.SamToFastq", "picard.util.ScatterIntervalsByNs", "picard.sam.SortSam", "picard.vcf.SortVcf", "picard.sam.SplitSamByLibrary",
+                      "picard.vcf.UpdateVcfSequenceDictionary", "picard.vcf.VcfFormatConverter", "picard.illumina.MarkIlluminaAdapters", "picard.vcf.SplitVcfs",
+                      "picard.sam.ValidateSamFile", "picard.sam.ViewSam", "picard.vcf.VcfToIntervalList"]
+
+task documentInit(dependsOn: 'classes') {
+    commandLineHtmlDir.delete()
+    commandLineHtmlDir.mkdirs()
+}
+
+task documentAll
+
+task documentStandardOptions(dependsOn: 'documentInit', type: JavaExec) {
+    main = 'picard.cmdline.CreateHtmlDocForStandardOptions'
+    classpath = sourceSets.main.runtimeClasspath
+    standardOutput = new FileOutputStream(standardOptionsFile)
+    documentAll.dependsOn "documentStandardOptions"
+}
+
+ext.lastDocTask = null
+ext.currentDocTask = null
+
+commandClasses.each { mainClass ->
+    currentDocTask = task "document_${mainClass}"(dependsOn: 'documentInit', type: JavaExec) {
+        main = 'picard.cmdline.CreateHtmlDocForProgram'
+        classpath = sourceSets.main.runtimeClasspath
+        args mainClass
+        standardOutput = new FileOutputStream(new File(commandLineHtmlDir, mainClass.substring(mainClass.lastIndexOf(".") + 1) + ".html"))
+        doLast {
+            usageFile.append("{% include ${mainClass.substring(mainClass.lastIndexOf(".") + 1) + ".html"} %}")
+            usageFile.append(System.getProperty("line.separator"))
+            sidebarFile.append("<li><a href=\"command-line-overview.html#${mainClass.substring(mainClass.lastIndexOf(".") + 1)}\">${mainClass.substring(mainClass.lastIndexOf(".") + 1)}</a>")
+            sidebarFile.append(System.getProperty("line.separator"))
+        }
+    }
+    if (lastDocTask != null) currentDocTask.dependsOn lastDocTask
```

Comment:
I'm assuming this ordering is important?   It seems a bit awkward the way it's done right now.  Maybe it would be cleaner to use `.eachwithindex` instead of the global properties.  Maybe not though.